src/olympia/devhub/tasks.py

```diff
@@ -267,6 +271,52 @@ def handle_file_validation_result(results, file_id, annotate=True):
     return FileValidation.from_json(file_, results)
 
 
+def annotate_new_legacy_addon_restrictions(results):
+    """
+    Annotate validation results to restrict uploads of new legacy
+    (non-webextension) add-ons if specific conditions are met.
+    """
+    metadata = results.get('metadata', {})
+    target_apps = metadata.get('applications', {})
+    target_firefox_version = (
+        target_apps.get('firefox', {}).get('max', None) or
```

Comment:
Actually, a new submission having maxversions like Firefox Android 54 and Desktop 51 would pass this. Admittedly, a little of an edge-case...